keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,184 @@
+    respectively. We then sample offsets for height and width and clip them such
+    that the cropped area does not exceed image boundaries. Finally we do the
+    actual cropping operation and resize the image to `target_size`.
+
+    Args:
+        target_size: A tuple of two integers used as the target size to ultimately crop
+            images to.
+        area_factor: (Optional) A tuple of two floats, a single float or
+            `keras_cv.FactorSampler`. The ratio of area of the cropped part to
+            that of original image is sampled using this factor. Represents the
+            lower and upper bounds for the area relative to the original image
+            of the cropped image before resizing it to `target_size`.
+            `target_size`.  Defaults to (0.08, 1.0).
+        aspect_ratio_factor: (Optional) A tuple of two floats, a single float or
+            `keras_cv.FactorSampler`. Aspect ratio means the ratio of width to
+            height of the cropped image. In the context of this layer, the aspect ratio
+            sampled represents a value to distort the aspect ratio by.
+            Represents the lower and upper bound for the aspect ratio of the
+            cropped image before resizing it to `target_size`. Defaults to
+            (3./4., 4./3.).
+        interpolation: (Optional) A string specifying the sampling method for
+            resizing. Defaults to "bilinear".
+        seed: (Optional) Used to create a random seed. Defaults to None.
+    """
+    def __init__(
+        self,
+        target_size,
+        area_factor=(0.08, 1.0),
+        aspect_ratio_factor=(3. / 4., 4. / 3.),
+        interpolation="bilinear",
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, **kwargs)
+
+        self.target_size = target_size
+
+        aspect_ratio_factor = aspect_ratio_factor or (3. / 4., 4. / 3.)
+        if isinstance(aspect_ratio_factor, tuple):
+            min_aspect_ratio = min(aspect_ratio_factor)
+            max_aspect_ratio = max(aspect_ratio_factor)
+        elif isinstance(aspect_ratio_factor, core.FactorSampler):
+            pass
+        else:
+            raise ValueError(
+                "Expected `aspect_ratio` to be tuple or FactorSampler. Received "
+                f"RandomResizedCrop(aspect_ratio_factor={aspect_ratio_factor})."
+            )
+
+        self.aspect_ratio_factor = preprocessing.parse_factor(
+            aspect_ratio_factor,
+            min_value=min_aspect_ratio,
+            max_value=max_aspect_ratio,
+            param_name="aspect_ratio_factor",
+            seed=seed,
+        )
+        self.area_factor = preprocessing.parse_factor(
+            area_factor,
+            max_value=1.0,
+            param_name="area_factor",
+            seed=seed,
+        )
+
+        self.interpolation = interpolation
+        self.seed = seed
+
+        if area_factor == 0.0 and aspect_ratio_factor == 0.0:
```

Comment:
I think that an area factor of 0.0 is actually nonsense, right?  A no-op should be 1.0, and a no-op for aspect ratio is also 1.0.